apps/antalmanac/src/components/Calendar/CalendarRoot.tsx

```
@@ -70,7 +78,157 @@ export const ScheduleCalendar = memo(() => {
         showFinalsSchedule,
     ]);
 
-    const events = getEventsForCalendar();
+    useEffect(() => {
```

This is a nice add-on. I would also look into the instance where the user adds/deletes an item from the schedule. For consistency I think we should cache the change as well so that it's reflected on the next reload.

Repro: Open schedule -> remove or add a lec or dis -> refresh 

Expected: 
- The loading items should reflect the most recent changes 

Actual:
- Since the changes weren't cached we default to the random preset